s2a/src/main/java/io/grpc/s2a/S2AChannelCredentials.java

```diff
@@ -106,15 +108,9 @@ public Builder setLocalUid(String localUid) {
       return this;
     }
 
-    /** Sets the credentials to be used when connecting to the S2A. */
-    @CanIgnoreReturnValue
-    public Builder setS2AChannelCredentials(ChannelCredentials s2aChannelCredentials) {
-      this.s2aChannelCredentials = s2aChannelCredentials;
-      return this;
-    }
-
-    public ChannelCredentials build() {
+    public ChannelCredentials build() throws IOException {
       checkState(!isNullOrEmpty(s2aAddress), "S2A address must not be null or empty.");
```

Comment:
Both of these are already checked in `newBuilder()`. They should be verified when being set, not during build(), so even if we add set methods later, we wouldn't want them here.

Reply:
Done in https://github.com/grpc/grpc-java/pull/11544/commits/36222a07ae68335b4f17666d2e8d437aeed8a4c1